Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/FitGaussianPeaksTest.py

```diff
@@ -514,6 +532,54 @@ def test_algorithm_does_not_throw_an_error_when_no_valid_peaks_fitted(self):
         self.assertEqual(mtd["peak_table"].rowCount(), 0)
         self.assertEqual(mtd["refit_peak_table"].rowCount(), 0)
 
+    def test_algorithm_uses_right_fit_window(self):
+        ws = CreateWorkspace([0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22] * 2,
+                             [0, 0, 0, 0, 1, 7, 0, 0, 0, 0, 10, 7] + [0] * 12, NSpec=2)
+
+        table = CreateEmptyTableWorkspace()
+        table.addColumn("float", "Centre")
+        table.addRow([20])
+        with mock.patch('plugins.algorithms.WorkflowAlgorithms.FitGaussianPeaks.FitGaussianPeaks.'
+                        'estimate_single_parameters') as mock_estimate_params:
+            mock_estimate_params.return_value = None
+            FitGaussianPeaks(InputWorkspace=ws,
+                             PeakGuessTable=table,
+                             EstimateFitWindow=False,
+                             FitWindowSize=11)
+            correct_x_val = np.array([0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22])
+            correct_y_val = np.array([0, 0, 0, 0, 1, 7, 0, 0, 0, 0, 10, 7])
+
+            arguements = mock_estimate_params.call_args_list[0][0]
+            np.testing.assert_equal(arguements[0], correct_x_val)
+            np.testing.assert_equal(arguements[1], correct_y_val)
+            np.testing.assert_equal(arguements[2], 10)
```

Comment:
why 10 and 5?

Reply:
10 is the index of the centre of the peak to fit, 5 relates to the FitWindowSize it is (FitWindowSize - 1)/2 as this function requires it to be in this form

Reply:
at the moment it just seems to be magic. Could you add an explanation to the code?